node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -434,11 +435,18 @@ mod tests {
             "node_configurator",
             "server_initializer_collected_params_can_read_parameters_from_config_file",
         );
+        let chain_specific_dir = add_chain_specific_directories(Chain::PolyMainnet, &home_dir);
         {
-            let mut config_file = File::create(home_dir.join("config.toml")).unwrap();
-            config_file
-                .write_all(b"dns-servers = \"111.111.111.111,222.222.222.222\"\n")
-                .unwrap();
+            let created_dir = create_dir_all(&chain_specific_dir);
```

Comment:
Please use the `.expect()` here too, as advised previously. It will shrink the if into a single-line call.